src/communication/communication.service.ts

```diff
@@ -0,0 +1,171 @@
+import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
+import { PrismaService } from '../utils/database/prisma.service';
+import { createEventDto, updateEventDto, createAnnouncementCommentDto } from './dto/communication.dto';
+
+@Injectable()
+export class CommunicationService {
+  constructor(private prisma: PrismaService) {}
+
+  async updateEvent(eventId: string, dto: updateEventDto)  {
+    const event = await this.prisma.event.findUnique({
+      where: {
+        id: eventId,
+      },
+    });
+
+    if (!event) {
+      throw new NotFoundException('Event not found');
+    }
+
+    return this.prisma.event.update({
+      where: {
+        id: eventId,
+      },
+      data: dto,
+    });
+  }
+
+  async createEvent(dto: createEventDto) {
+    try {
+      return await this.prisma.event.create({
+        data: dto,
+      });
+    } catch (error) {
+      throw new BadRequestException('Failed to create the event. Details: ' + error.message);
+    }
+  }
+
+  async deleteEvent(eventId: string) {
+    const event = await this.prisma.event.findUnique({
+      where: {
+        id: eventId,
+      },
+    });
+
+    if (!event) {
+      throw new NotFoundException('Event not found');
+    }
+
+    return this.prisma.event.delete({
+      where: {
+        id: eventId,
+      },
+    });
+  }
+  async getEvent(eventId: string) {
+    const event = await this.prisma.event.findUnique({
+      where: {
+        id: eventId,
+      },
+    });
+
+    if (!event) {
+      throw new NotFoundException('Event not found');
+    }
+
+    return event;
+  }
+
+  async getAllEvents() {
+    return this.prisma.event.findMany();
+  }
+
+  async deleteAnnouncementComment(announcementId: string, commentId: string) {
+    const announcement = await this.prisma.announcement.findUnique({
+      where: {
+        id: announcementId,
```

Comment:
Should add a check if there in an announcement with the specified ID AND commend id.

```suggestion
        id: announcementId,
        comments: {
            id: commentId
        }
```